src/bm_sim/switch.cpp

```diff
@@ -80,6 +80,8 @@ SwitchWContexts::start_and_return() {
     Logger::get()->error(
         "The switch was started with no P4 and config swap is disabled");
   }
+  // The lock is held until the config is loaded to avoid race conditions with   
+  // starting the PIGrpcServerRunV2 (P4RT server).
```

Comment:
That's not really the complete picture, and this part of the code really has no knowledge of P4Runtime.

I think something like this is more appropriate:

> We keep holding the lock as the start_and_return_ callback which is implemented by targets may access the P4 config for some validations.